src/main/java/com/bouncestorage/swiftproxy/v1/AccountResource.java

```diff
@@ -111,42 +120,61 @@ public Response headAccount(@NotNull @PathParam("account") String account,
                                        @HeaderParam("X-Auth-Token") String authToken,
-                                       String objectsList) throws JsonProcessingException {
-        if (!bulkDelete) {
+                                       @Context Request request) throws JsonProcessingException {
+        if (bulkDelete == null) {
             // TODO: Currently this will match the account delete request as well, which we do not implement
             throw new WebApplicationException(Response.Status.NOT_IMPLEMENTED);
         }
 
-        if (objectsList == null) {
-            return new BulkDeleteResult();
-        }
-        String[] objects = objectsList.split("\n");
         BlobStore blobStore = getBlobStore(authToken).get();
         if (blobStore == null) {
             throw new WebApplicationException(Response.Status.BAD_REQUEST);
         }
+
+        BufferedReader in = new BufferedReader(new InputStreamReader(request.getInputStream()));
+        String line;
+        ArrayList<String> objects = new ArrayList<>();
+
+        boolean isTransient = blobStore.getContext().unwrap().getId().equals("transient");
+        try {
+            while ((line = in.readLine()) != null) {
+                if (isTransient) {
+                    // jclouds does not escape things correctly
+                    line = Strings2.urlEncode(line, skipPathEncoding);
+                }
+                objects.add(line);
+            }
+        } catch (IOException e) {
+            throw propagate(e);
+        }
+
         BulkDeleteResult result = new BulkDeleteResult();
         for (String objectContainer : objects) {
             try {
-                if (!objectContainer.startsWith("/")) {
-                    result.errors.add(objectContainer);
-                    continue;
+                if (objectContainer.startsWith("/")) {
+                    objectContainer = objectContainer.substring(1);
                 }
-                int separatorIndex = objectContainer.indexOf("/", 1);
+                int separatorIndex = objectContainer.indexOf('/');
                 if (separatorIndex < 0) {
                     blobStore.deleteContainer(objectContainer.substring(1));
                     result.numberDeleted += 1;
                     continue;
                 }
-                String container = objectContainer.substring(1, separatorIndex);
+                String container = objectContainer.substring(0, separatorIndex);
                 String object = objectContainer.substring(separatorIndex + 1);
-                blobStore.removeBlob(container, object);
-                result.numberDeleted += 1;
+
+                if (!blobStore.blobExists(container, object)) {
+                    result.numberNotFound += 1;
+                } else {
+                    blobStore.removeBlob(container, object);
+                    result.numberDeleted += 1;
+                }
```

Comment:
Shouldn't we call `blobStore.removeBlobs` to preserve the user intent?  We can fake these statistics.